File: frontend/src/microservices/tour-group/objects/TourGroupSummaryResponse.ts
// 本文件定义 tour-group 模块的 `TourGroupSummaryResponse`，作为摘要响应数据并提供 JSON 编解码。

export type TourGroupSummaryResponse = {
  groupId: string
  organizerUserId: string
  title: string
  description: string
  destination: string
  startDate: string
  endDate: string
  capacity: number
  usedCapacity: number
  coverImageUrl: string | null
  tags: string[]
  memberCount: number
  pendingSelectionCount: number
  confirmedSelectionCount: number
  convertedOrderCount: number
  status: string
  createdAt: string
}
export const tourGroupSummaryResponseFromJson = (json: string): TourGroupSummaryResponse =>
  JSON.parse(json) as TourGroupSummaryResponse

export const tourGroupSummaryResponseToJson = (value: TourGroupSummaryResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/TourGroupSummaryResponse.scala
// TourGroupPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TourGroupSummaryResponse(
    groupId: String,
    organizerUserId: String,
    title: String,
    description: String,
    destination: String,
    startDate: String,
    endDate: String,
    capacity: Int,
    usedCapacity: Int,
    isFull: Boolean,
    coverImageUrl: Option[String],
    tags: List[String],
    memberCount: Int,
    activeTravelerCount: Int,
    pendingSelectionCount: Int,
    confirmedSelectionCount: Int,
    convertedOrderCount: Int,
    status: String,
    createdAt: String
)
object TourGroupSummaryResponse:
  given sourceEncoder: Encoder[TourGroupSummaryResponse] = deriveEncoder
  given sourceDecoder: Decoder[TourGroupSummaryResponse] = deriveDecoder
